lib/components/secure-payment-component.js

```diff
@@ -117,6 +117,67 @@ export default class SecurePaymentComponent extends AsyncBaseContainer {
 		return await this.submitPaymentDetails();
 	}
 
+	async clickCouponButton() {
+		await driverHelper.waitTillPresentAndDisplayed(
+			this.driver,
+			By.css( 'button.cart__toggle-link' )
+		);
+
+		// 2 elements have this locator on the page including the mobile element
+		let elements = await this.driver.findElements( By.css( 'button.cart__toggle-link' ) );
```

Comment:
I think it would be best here to have two selectors, one for mobile and another for desktop which both return a single element (something like `.payment-box__content button.cart__toggle-link`) - that way there's no need to loop - it's easy to know whether the test is mobile or desktop (`driverManager.currentScreenSize();`)